cms/djangoapps/contentstore/tasks.py

```diff
@@ -47,10 +48,224 @@
+    """
+    store = modulestore()
+    kwargs = {
+        'force_update': force_update,
+        'commit': commit
+    }
+    if all_courses:
+        course_keys = [course.id for course in store.get_course_summaries()]
+    else:
+        course_keys = [CourseKey.from_string(id) for id in course_ids]
+    try:
+        tasks = [
+            async_migrate_transcript.s(
+                unicode(course_key),
+                **kwargs
+            ) for course_key in course_keys
+        ]
+        callback = task_status_callback.s()
+        status = chord(tasks)(callback)
+        for res in status.get():
+            LOGGER.info("Migration result: %s", '\n'.join(res))
+    except Exception as exc:
+        LOGGER.exception('Exception: %r', text_type(exc))
+
+
+@task
+def task_status_callback(results):
+    """
+        Callback for collating the results of chord.
+    """
+    return results
+
+
+@task(base=PersistOnFailureTask)
+def async_migrate_transcript(course_key, **kwargs):
+    """
+        Migrates the transcripts of all videos in a course as a new celery task.
+    """
+    try:
+        if not modulestore().get_course(CourseKey.from_string(course_key)):
+            raise KeyError(u'Invalid course key: ' + unicode(course_key))
+        force_update = kwargs['force_update']
+        commit = kwargs['commit']
+        sub_tasks = []
+
+        LOGGER.info("[Transcript migration] process for course %s started", course_key)
+        all_videos = get_videos_from_store(CourseKey.from_string(course_key))
+
+        for video in all_videos:
+            other_lang_transcripts = video.transcripts
+            english_transcript = video.sub
+            LOGGER.info("[Transcript migration] process for video %s started", video.location)
+            all_language_transcripts = dict({'en': video.sub}, **other_lang_transcripts) if english_transcript\
+                else other_lang_transcripts
+            for lang, name in all_language_transcripts.items():
+                transcript_already_present = is_transcript_available(video.edx_video_id, lang)
+                LOGGER.info("Already pushed transcript of language %s found: %s ",
+                            lang,
+                            transcript_already_present
+                            )
+                if transcript_already_present and force_update:
+                    sub_tasks.append(async_migrate_transcript_subtask.s(
+                        video, lang, name, True, **kwargs
+                    ))
+                elif not transcript_already_present:
+                    sub_tasks.append(async_migrate_transcript_subtask.s(
+                        video, lang, name, False, **kwargs
+                    ))
+            LOGGER.info("[Transcript migration] process for video %s ended", video.location)
+        callback = task_status_callback.s()
+        status = chord(sub_tasks)(callback)
```

Comment:
@nasthagiri I have used a celery task for each video inside a celery task for each course. Do you think this could potentially create too many celery tasks, which could potentially swamp the task queue? The benefit of a 'task per video' could be a more granular failure tracking and retry.

Reply:
@irfanuddinahmad That should be fine given our other recent usage of celery tasks:
* we create a celery task for each learner that we send an email to via ACE
* we create a celery task for each problem submission by a learner

However, you should let devops know so they are aware in case they see spikes in tasks at certain times.  Also, the team will want to monitor this the first time you roll it out.

Things to consider:
* Is there a time_limit placed on the task?  If your task is calling the network, you'll want to timeout appropriately (and exponentially backoff) in case of a persistent network failure.

* Are you using our [celery-utils|https://github.com/edx/edx-celeryutils], which provides functionality for logging and persistent retries?  You may find this useful if you want to persist failures (after celery's retries give up) and automatically retry them via a Jenkins job.

You can see some of this in action in the [grades tasks code|https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/tasks.py].

Reply:
@nasthagiri Thanks. Very useful feedback (as usual! :) )
I am already using the PersistOnFailureTask ... let me try the 
celeryutils.chordable_django_backend and LoggedPersistOnFailureTask also